src/main/scala/mesosphere/marathon/core/task/bus/MesosTaskStatus.scala

```diff
@@ -18,9 +18,15 @@ object MesosTaskStatus {
 
   val WontComeBack: Set[TaskStatus.Reason] = TaskStatus.Reason.values().toSet.diff(MightComeBack)
 
+  def wontComeBack(status: TaskStatus): Boolean = {
+    // FIXME (gkleiman): comparing against a string is ugly and fragile, we shouldn't do it, but there's no better way
+    // for now
+    WontComeBack(status.getReason) || status.getMessage.startsWith("Reconciliation: Task is unknown to the")
```

Comment:
This is correct but not sufficient. The `TemporarilyUnreachable` extractor must also consider the message if the reason is `TASK_LOST`.

Besides, The `MightComeBack` set includes `REASON_RECONCILIATION`, which is not completely correct any more.